public/app/plugins/datasource/grafana-azure-monitor-datasource/components/LogsQueryEditor/LogsQueryEditor.tsx

```diff
@@ -23,6 +23,26 @@ const LogsQueryEditor: React.FC<LogsQueryEditorProps> = ({
   onChange,
   setError,
 }) => {
+  const migrateWorkspaceQueriesToResourceQueries = useCallback(async () => {
+    if (query.azureLogAnalytics.workspace !== undefined && !query.azureLogAnalytics.resource) {
+      const resourceURI = await datasource.resourcePickerData.getResourceURIFromWorkspace(
+        query.azureLogAnalytics.workspace
+      );
+      onChange({
+        ...query,
+        azureLogAnalytics: {
+          ...query.azureLogAnalytics,
+          resource: resourceURI,
+          workspace: undefined,
+        },
+      });
+    }
+  }, [datasource, onChange, query]);
+
+  useEffect(() => {
+    migrateWorkspaceQueriesToResourceQueries();
+  }, [query, migrateWorkspaceQueriesToResourceQueries]);
```

Comment:
Actually, do we think we could extract this out into a seperate function/hook, like `useApplyMigrations()`?